File: Dante1/Module1.05/Zadanie5.12/main.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main()
{
    char data[101];
    printf("Enter: ");

    long long counter = 0;
    scanf("%100[^\n]", data);
    getchar();

    for(size_t i = 0; i < strlen(data); i++)
    {
        if(data[i] == '-' || data[i] == '+')
        {
            if(data[i + 1] == '-' || data[i + 1] == '+')
            {
                printf("Incorrect input");
                return 1;
            }
        }
        if(isalpha(data[i]))
        {
            printf("Incorrect input");
            return 1;
        }
    }

    long long num = 0;
    int tempshit = '+';

    for (size_t i = 0; i < strlen(data); i++)
    {
        if (isdigit(data[i]))
        {
            num = num * 10 + (data[i] - '0');
        }
        else if (data[i] == '+' || data[i] == '-')
        {
            if (tempshit == '+')
            {
                counter = counter + num;
            }
            else if (tempshit == '-')
            {
                counter = counter - num;
            }

            num = 0;
            tempshit = (int)data[i];
        }
    }

    if (tempshit == '+')
    {
        counter = counter + num;
    }
    else if (tempshit == '-')
    {
        counter = counter - num;
    }


    printf("\n%lld", counter);

    return 0;
}

File: Dante1/Module1.05/Zadanie5.12/main.rs
fn main() {
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).expect("Error");
    let data = input.as_str();

    println!("Enter: ");

    for i in 0..data.len() - 1 {
        if data.chars().nth(i) == Option::from('-') || data.chars().nth(i) == Option::from('+') {
            if data.chars().nth(i + 1) == Option::from('-') || data.chars().nth(i + 1) == Option::from('+') {
                panic!("Incorrect");
            }
        }
        if data.chars().nth(i).expect("Error").is_ascii_alphanumeric() {
            panic!("Incorrect");
        }
    }

    let mut counter: u64 = 0;
    let mut number: u64 = 0;
    let mut temp: char = '+';

    for i in 0..data.len() {
        if data.chars().nth(i).expect("Error").is_ascii_alphanumeric() {
            number = number * 10 + (data.chars().nth(i).expect("Error") as u8 - '0' as u8) as u64;
        }
        else if data.chars().nth(i) == Option::from('-') || data.chars().nth(i) == Option::from('+') {
            if temp == '+' {
                counter = counter + number;
            }
            else if temp == '-' {
                counter = counter - number;
            }

            number = 0;
            temp = data.chars().nth(i).expect("Error");
        }
    }

    if temp == '+' {
        counter = counter + number;
    }
    else if temp == '-' {
        counter = counter - number;
    }

    println!("{} ", counter);
}
